libcodechecker/server/config_db_model.py

```diff
@@ -0,0 +1,93 @@
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+"""
+SQLAlchemy ORM model for the product configuration database.
+"""
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from sqlalchemy import *
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import *
+
+CC_META = MetaData(naming_convention={
+    "ix": 'ix_%(column_0_label)s',
+    "uq": "uq_%(table_name)s_%(column_0_name)s",
+    "ck": "ck_%(table_name)s_%(column_0_name)s",
+    "fk": "fk_%(table_name)s_%(column_0_name)s_%(referred_table_name)s",
+    "pk": "pk_%(table_name)s"
+})
+
+# Create base class for ORM classes.
+Base = declarative_base(metadata=CC_META)
+
+
+class DBVersion(Base):
+    __tablename__ = 'db_version'
+    # TODO: constraint, only one line in this table!
+    major = Column(Integer, primary_key=True)
+    minor = Column(Integer, primary_key=True)
+
+    def __init__(self, major, minor):
+        self.major = major
+        self.minor = minor
+
+
+# class Superuser(Base):
+#     __tablename__ = 'superusers'
+#
+#     id = Column(Integer, autoincrement=True, primary_key=True)
+#     name = Column(String, nullable=False)
+#     is_group = Column(Boolean, nullable=False)
+#
+#     def __init__(self, name, is_group):
+#         self.name = name
+#         self.is_group = is_group
+
+
+class Product(Base):
+    __tablename__ = 'products'
+
+    __table_args__ = (
+        UniqueConstraint('endpoint'),
```

Comment:
I would rather have an ID as primary key in the `product_admin` table, and various other permission tables that are upcoming. Strings as primary keys are a problematic subject... for example, MSSQL does not allow foreign keying onto a string (personal experience from half a year ago).